api/storage_service.cc

```diff
@@ -1369,7 +1391,7 @@ void set_snapshot(http_context& ctx, routes& r, sharded<db::snapshot_ctl>& snap_
         } else if (quarantine_mode_str == "ONLY") {
             opts.quarantine_operation_mode = sstables::compaction_type_options::scrub::quarantine_mode::only;
         } else {
-            throw std::invalid_argument(fmt::format("Unknown argument for 'quarantine_mode' parameter: {}", quarantine_mode_str));
+            throw httpd::bad_param_exception(fmt::format("Unknown argument for 'quarantine_mode' parameter: {}", quarantine_mode_str));
```

Comment:
Same here, need to see that nodetool works

Reply:
nodetool currently doesn't support the quarantine mode at all.
It supports `scrub_mode` and there's a dtest that exercises `scrub_mode=SEGREGATE` that still works.

Reply:
the only concern that I had was if when you run nodetool, you will get the error you expect. there are many moving parts in the way. if it's fine, great

Reply:
I verified that nodetool returns the error I expect now, and doesn't break.